pennylane/templates/subroutines/qsvt.py

```diff
@@ -455,9 +455,16 @@ def compute_matrix(*args, **kwargs):
 
 def _qsp_to_qsvt(angles):
     r"""Converts qsp angles to qsvt angles."""
+    num_angles = len(angles)
     new_angles = qml.math.array(copy.copy(angles))
-    new_angles[0] += 3 * np.pi / 4
-    new_angles[-1] -= np.pi / 4
+    indicies = qml.math.convert_like(np.arange(num_angles), angles)
 
-    new_angles[1:-1] += np.pi / 2
+    update_vals = np.zeros(num_angles)
```

Comment:
Empty is better since you're writing the entire array anyways:
```suggestion
    update_vals = np.empty(num_angles)
```

Reply:
Good call.